client_plugin/drivers/shared/shared_driver.go

```diff
@@ -62,6 +86,18 @@ func NewVolumeDriver(cfg config.Config, mountDir string) *VolumeDriver {
 	}
 	d.dockerd = cli
 
+	// Pull Samba image
+	log.Infof("Pulling Samba server image")
```

Comment:
Yes, I think pull on install is reasonable, also guarantees no out-of-disk-space failures on runtime.